src/style/__tests__/colors.spec.js

```diff
@@ -123,4 +124,39 @@ describe('style/Colors', () => {
       expect(palette).toEqual(['#661A66', '#8F248F', '#B82EB7', '#D148D1', '#DB71DB', '#E699E6', '#F0C2F0', '#FFE5FF']);
     });
   });
+
+  describe('getSystemColorByHex', () => {
+    it('should return system color from valid colors in the array', () => {
+      expect(uut.getSystemColorByHex('#FFFFFF', SYSTEM_COLORS)).toEqual('white');
+      expect(uut.getSystemColorByHex('#A6ACB1', SYSTEM_COLORS)).toEqual('grey40');
+      expect(uut.getSystemColorByHex('#20303C', SYSTEM_COLORS)).toEqual('grey10');
+      expect(uut.getSystemColorByHex('#000000', SYSTEM_COLORS)).toEqual('black');
+    });
+
+    it('should return undefined if color not include in validColors', () => {
+      expect(uut.getSystemColorByHex('#116DFF', SYSTEM_COLORS)).toEqual(undefined);
+      expect(uut.getSystemColorByHex('#00A87E', SYSTEM_COLORS)).toEqual(undefined);
+      expect(uut.getSystemColorByHex('#FFC50D', SYSTEM_COLORS)).toEqual(undefined);
+      expect(uut.getSystemColorByHex('#FB6413', SYSTEM_COLORS)).toEqual(undefined);
+    });
+
+    it('should return premium, fit, dark colors when validColors include premium, fit, dark', () => {
+      expect(uut.getSystemColorByHex('#116DFF', [...SYSTEM_COLORS, 'blue'])).toEqual('blue30');
+      expect(uut.getSystemColorByHex('#00A87E', [...SYSTEM_COLORS, 'green'])).toEqual('green30');
+      expect(uut.getSystemColorByHex('#FFC50D', [...SYSTEM_COLORS, 'yellow'])).toEqual('yellow30');
+      expect(uut.getSystemColorByHex('#FB6413', [...SYSTEM_COLORS, 'orange'])).toEqual('orange30');
+    });
+
+    it('should return system color without validColors array', () => {
```

Comment:
I'd change the text a bit, you can mention here that without the array it searches the color in all the colors